### ISSUE TITLE
ITIR to SDFG: Fixed sized dimensiosn.

### ISSUE THREAD

**Issue by philip-paul-mueller** [GridTools/gt4py#1412]
Currently all array sizes are turned into symbols during the translation from ITIR to SDFG.
While this is correct for a lot of sizes, some are actually known and should not be symbols.
An example of such a size would be `E2VDim` which is always two, but currently it is a symbol that must be passed as an argument to the function.
Another issue with this is, that it forces DaCe to dynamically allocate of arrays of such length (this happens for example in `list_get` and probably other places).

My suggestion would be to change the code generator such that such sizes/dimensions are no longer considered variable but have fixed sizes.

- This issue is the root cause of this [DaCe issue](https://github.com/spcl/dace/issues/1500).
- I discovered the issue during my work on the stencil.

**Comment by edopao** [comment]
Good point. We can set symbols as constants after we have built the SDFG. There is actually a TODO for this, although the comment is outdated:
```
symbols: dict[str, int] = {}
device = dace.DeviceType.GPU if on_gpu else dace.DeviceType.CPU
sdfg = autoopt.auto_optimize(sdfg, device, symbols=symbols, use_gpu_storage=on_gpu)
```

We could set the shape/strides for the connectivity tables as constants there. The implementation of `auto_optimize` will set the constants:
```
    if symbols:
        # Specialize for all known symbols
        ...
        sdfg.specialize(known_symbols)
```

**Comment by havogt** [comment]
Just for reference: we discussed that itir.list is a dynamically sized array with a static max size. That's probably how it should be implemented in the itir to sdfg lowering.

**Comment by edopao** [comment]
This issue should be fixed by PR#1430.
